diff --git a/addons/sale/sale.py b/addons/sale/sale.py
index 4e7598f..88fdcd5 100644
--- a/addons/sale/sale.py
+++ b/addons/sale/sale.py
@@ -1039,7 +1039,7 @@ class sale_order_line(osv.osv):
         return {'value': value}

     def create(self, cr, uid, values, context=None):
-        if values.get('order_id') and values.get('product_id') and  any(f not in values for f in ['name', 'price_unit', 'type', 'product_uom_qty', 'product_uom']):
+        if values.get('order_id') and values.get('product_id') and  any(f not in values for f in ['name', 'price_unit', 'product_uom_qty', 'product_uom']):
             order = self.pool['sale.order'].read(cr, uid, values['order_id'], ['pricelist_id', 'partner_id', 'date_order', 'fiscal_position'], context=context)
             defaults = self.product_id_change(cr, uid, [], order['pricelist_id'][0], values['product_id'],
                 qty=float(values.get('product_uom_qty', False)),
